### DISCUSSION TITLE
How to add healthcheck for Bunny without overwhelming logs?

### DISCUSSION THREAD

**Discussion by PoroshkinaVV** [ruby-amqp/bunny#689] [General]
I have an app within k8s. For my probes I want to check if bunny is up and ready to work.

My code for healtcheck: 
```
client = ::Bunny.new(config)
client.start
client.close
```

Probes starts every 10s and I get really huge amount of info messages, that has no useful information for my app, caused by line `client.start`. 
This leads me to the line: https://github.com/ruby-amqp/bunny/blob/8e79ee852d16a5d10db5b5717712a066f32ca20e/lib/bunny/session.rb#L1273

So I want to ask, is there a possibility to change it's log_level to debug from info? 
Or, maybe, there is some best practice, how to add healthcheck for Bunny, without need to start session?

**Comment by michaelklishin** [comment]
@PoroshkinaVV that message is not particularly important but the only reason why your get "a lot of `info` messages" is because you have chosen to open new connections with the default log level 6 times a minute per pod.

You can set it to `error` for those one-off connections without any changes to Bunny or other connection settings.

The Monitoring doc guide for RabbitMQ itself both discusses [health checks](https://www.rabbitmq.com/docs/monitoring#health-checks) and [recommended monitoring frequency](https://www.rabbitmq.com/docs/monitoring#monitoring-frequency).

**Comment by michaelklishin** [comment]
@PoroshkinaVV I've reduced the verbosity of that specific message to `debug` in `main`. You can override the log level used by these frequent one-off connections in your own code without depending on when the next release drops.

**Comment by PoroshkinaVV** [comment]
@michaelklishin thank you for your answer!!

I totally understand that.
> the only reason why your get "a lot of info messages" is because you have chosen to open new connections with the default log level 6 times a minute per pod.

That's why I asked about alternative ways of healthchecks: 

> how to add healthcheck for Bunny, without need to start session?

How can I do that? 

> You can set it to error for those one-off connections without any changes to Bunny or other connection settings.

This guides covers how to monitor rabbitmq itself. But I need to understand, if my application can connect to rabbitmq and don't have access to monitor Rabbitmq itself. 

> The Monitoring doc guide for RabbitMQ itself both discusses [health checks](https://www.rabbitmq.com/docs/monitoring#health-checks) and [recommended monitoring frequency](https://www.rabbitmq.com/docs/monitoring#monitoring-frequency).

**Comment by michaelklishin** [comment]
It makes sense monitor the number of unsuccessful authentication attempts and various operational metrics such as consumer delivery and acknowledgement rates. Bunny does not dictate how you do that.

I don't see much value in connecting every 30-60s, although at that rate, it won't do any harm unless you have tens or hundreds of instance of your app. [Connection churn](https://www.rabbitmq.com/docs/connections#high-connection-churn) should be avoided.
